cylc/flow/graph_parser.py

```diff
@@ -517,32 +518,33 @@ def _proc_dep_pair(
                 "Suicide markers must be"
                 f" on the right of a trigger: {left}")
 
+        # Check that parentheses match.
+        mismatch_msg = 'Mismatched parentheses in: "{}"'
+        if left and left.count("(") != left.count(")"):
+            raise GraphParseError(mismatch_msg.format(left))
+        if right.count("(") != right.count(")"):
+            raise GraphParseError(mismatch_msg.format(right))
```

Comment:
I'm outputting the entire thing might be helpful.
```suggestion
        mismatch_msg = 'Mismatched parentheses in: "{} => {}"'
        if left and left.count("(") != left.count(")"):
            raise GraphParseError(mismatch_msg.format(left, right))
        if right.count("(") != right.count(")"):
            raise GraphParseError(mismatch_msg.format(left, right))
```

Reply:
Sometimes there will be no arrow involved, however